aries_vcx_core/src/anoncreds/credx_anoncreds.rs

```diff
@@ -731,27 +731,59 @@ impl BaseAnonCreds for IndyCredxAnonCreds {
                     format!("Unknown referent: {}", reft),
                 ))?;
 
-            let _attr_name = requested_val.try_get("name")?;
-            let _attr_name = _attr_name.try_as_str()?;
-            let attr_name = _normalize_attr_name(_attr_name);
+            if let Some(_attr_names) = requested_val.get("names").and_then(|v| v.as_array()) {
+                let _attr_names = requested_val.try_get("names")?.as_array().ok_or_else(|| {
```

Comment:
It feels like a code smell because the `ok_or_else` err handling should be never reached. Perhaps would be cleaner to do a match on tuple of `(requested_val.get("names"), requested_val.get("name"))` and that would also sort out the invalid cases referred to in the comment above.